src/parse.rs

```diff
@@ -344,14 +344,31 @@ impl<'a> Bytes<'a> {
         s.bytes().enumerate().all(|(i, b)| self.bytes.get(i).map(|t| *t == b).unwrap_or(false))
     }
 
-    pub fn unsigned_integer<T>(&mut self) -> Result<T> where T: FromStr {
+    pub fn unsigned_integer<T: Num>(&mut self) -> Result<T> {
+        let base = if self.bytes.get(0) == Some(&b'0') {
+            match self.bytes.get(1).cloned() {
+                Some(b'x') => 16,
+                Some(b'b') => 2,
+                Some(b'o') => 8,
+                _ => 10,
```

Comment:
I wonder if we should `warn!()` here

Reply:
Just because the first character was a `0`, this doesn't have to be any control character (this could just be another digit). If it is an invalid character, trying to parse the integer fails, so this doesn't need special handling.